File: web/src/data/dtos/recipe/RecipeIngredientDto.ts
export class RecipeIngredientDto {
    sortOrder: number = 0;
    section: string | null = null;

    ingredientId: number = 0;
    ingredientName: string = '';

    amount: number | null = null;
    unit: string | null = null;
    amountText: string | null = null;
    note: string | null = null;

    constructor(obj: Partial<RecipeIngredientDto>) {
        Object.assign(this, obj);
    }
}

File: dtos/recipe_ingredient_dto.go
package dtos

type RecipeIngredientDto struct {
	SortOrder int     `json:"sortOrder"`
	Section   *string `json:"section,omitempty"`

	IngredientID   uint   `json:"ingredientId"`
	IngredientName string `json:"ingredientName"`

	Amount     *float64 `json:"amount,omitempty"`
	Unit       *string  `json:"unit,omitempty"`
	AmountText *string  `json:"amountText,omitempty"`
	Note       *string  `json:"note,omitempty"`
}
